### PR TITLE
test: Port all field columns query_tests to end-to-end tests

### PR DESCRIPTION
**Pull request by carols10cents**
Part of #6617.

I decided to rename these to "measurement fields" rather than "field columns" because that matches the gRPC API these tests are now using.

I also decided to extract them from the metadata tests file to a submodule now that there's more than one test.

I believe I have ported all the tests in the field_columns test file, so I've deleted it. I didn't find the test names especially clear, so I did change them, which I realize makes it a bit trickier to verify they've all been ported. I'll annotate and link them.

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+                assert_eq!(field.timestamp, ns_since_epoch + 4);
+            }
+            .boxed()
+        }),
+    )
+    .await
+}
+
+#[tokio::test]
+async fn field_columns_nonexistent_table_with_predicate() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:51]: Was `test_field_columns_no_predicate` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L52-L57

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        table_name: "NoSuchTable",
+        request: GrpcRequestBuilder::new().tag_predicate("state", "MA"),
+        expected_fields: vec![],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn field_columns_existing_table_with_predicate() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:63]: Was `test_field_columns_with_pred` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L59-L86

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                timestamp: 100_000,
+            },
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn field_columns_timestamp_range_predicate() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:91]: Was `test_field_columns_measurement_pred` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L88-L117

As discussed in chat, this test now has a different purpose given that its original formulation of having a `_measurement=name` predicate isn't valid.

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+                timestamp: 100_000,
+            },
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn field_columns_no_predicate_full_time_range_returns_zero_timestamp() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:119]: Was `test_field_columns_measurement_pred_all_time(` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L119-L147

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+                timestamp: 0,
+            },
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn field_columns_tag_and_timestamp_range_predicate() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:147]: Was `test_field_columns_with_ts_pred` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L149-L165

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+            r#type: FieldType::Float.into(),
+            timestamp: 250,
+        }],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn field_names() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:165]: Was `test_field_name_plan` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L167-L200

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+                timestamp: 1_000,
+            },
+        ],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn list_field_columns_all_time() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:198]: Was `list_field_columns_all_time` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L202-L216

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+            r#type: FieldType::Float.into(),
+            timestamp: 0,
+        }],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn list_field_columns_max_time_included() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:214]: Was `list_field_columns_max_time_included` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L218-L235

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+            r#type: FieldType::Float.into(),
+            timestamp: MAX_NANO_TIME,
+        }],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn list_field_columns_max_time_excluded() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:233]: Was `list_field_columns_max_time_excluded` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L237-L247

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+        // one less than max timestamp
+        request: GrpcRequestBuilder::new().timestamp_range(MIN_NANO_TIME + 1, MAX_NANO_TIME),
+        expected_fields: vec![],
+    })
+    .run()
+    .await;
+}
+
+#[tokio::test]
+async fn list_field_columns_with_periods() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:246]: Was `list_field_columns_with_periods` https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/query_tests/src/influxrpc/field_columns.rs#L249-L270

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs [hunk 11]
```diff
@@ -0,0 +1,303 @@
+use super::{run_data_test, InfluxRpcTest};
+use async_trait::async_trait;
+use data_types::{MAX_NANO_TIME, MIN_NANO_TIME};
+use futures::{prelude::*, FutureExt};
+use generated_types::measurement_fields_response::{FieldType, MessageField};
+use std::sync::Arc;
+use test_helpers_end_to_end::{DataGenerator, GrpcRequestBuilder, MiniCluster, StepTestState};
+
+#[tokio::test]
+async fn measurement_fields() {
```
Comment (carols10cents) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata/measurement_fields.rs:10]: This was the one test that was in influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata.rs: https://github.com/influxdata/influxdb_iox/blob/0249b5018e1b1b3018153f48d817b619fe5e3cc8/influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata.rs#L247-L286

Let me know if you think this is now sufficiently covered with the other tests and I'm happy to remove.

**Review by carols10cents — COMMENTED** [review]
Annotating where the tests came from

**Review by alamb — APPROVED** [review]
Thank you @carols10cents  -- I double checked that all the tests were ported correctly, and I also verified if I changed the expected output the tests fail, as expected 👍

**Comment by alamb** [comment]
Given the other work that is about to hit this code I took the liberty of merging this PR to reduce the need for conflicts

**Pull request merged**